component/loki/source/api/internal/lokipush/push_api_server.go

```diff
@@ -0,0 +1,240 @@
+package lokipush
+
+import (
+	"bufio"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/component/common/loki"
+	fnet "github.com/grafana/agent/component/common/net"
+	frelabel "github.com/grafana/agent/component/common/relabel"
+	"github.com/grafana/dskit/tenant"
+	"github.com/grafana/loki/pkg/loghttp/push"
+	"github.com/grafana/loki/pkg/logproto"
+	util_log "github.com/grafana/loki/pkg/util/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/relabel"
+	promql_parser "github.com/prometheus/prometheus/promql/parser"
+)
+
+type PushAPIServer struct {
+	logger       log.Logger
+	serverConfig *fnet.ServerConfig
+	server       *fnet.TargetServer
+	handler      loki.EntryHandler
+
+	rwMutex       sync.RWMutex
+	labels        model.LabelSet
+	relabelRules  []*relabel.Config
+	keepTimestamp bool
+}
+
+func NewPushAPIServer(logger log.Logger,
+	serverConfig *fnet.ServerConfig,
+	handler loki.EntryHandler,
+	registerer prometheus.Registerer,
+) (*PushAPIServer, error) {
+
+	s := &PushAPIServer{
+		logger:       logger,
+		serverConfig: serverConfig,
+		handler:      handler,
+	}
+
+	srv, err := fnet.NewTargetServer(logger, "agent_loki_source_api", registerer, serverConfig)
```

Comment:
As I mentioned, let's drop the metrics prefix.
```suggestion
	srv, err := fnet.NewTargetServer(logger, "loki_source_api", registerer, serverConfig)
```